core/trino-main/src/main/java/io/trino/sql/planner/AnonymizePlanRewriter.java

```diff
@@ -0,0 +1,785 @@
+                aggregation.getOrderingScheme().map(this::anonymize),
+                aggregation.getMask().map(this::anonymize));
+    }
+
+    private OrderingScheme anonymize(OrderingScheme scheme)
+    {
+        return new OrderingScheme(
+                anonymize(scheme.getOrderBy()),
+                anonymize(scheme.getOrderings()));
+    }
+
+    private ListMultimap<Symbol, Symbol> anonymize(ListMultimap<Symbol, Symbol> symbols)
+    {
+        ImmutableListMultimap.Builder<Symbol, Symbol> anonymizedSymbols = ImmutableListMultimap.builder();
+        symbols.forEach((key, val) -> anonymizedSymbols.put(anonymize(key), anonymize(val)));
+        return anonymizedSymbols.build();
+    }
+
+    private List<Symbol> anonymize(List<Symbol> symbols)
+    {
+        return symbols.stream()
+                .map(this::anonymize)
+                .collect(toImmutableList());
+    }
+
+    private Set<Symbol> anonymize(Set<Symbol> symbols)
+    {
+        return symbols.stream()
+                .map(this::anonymize)
+                .collect(toImmutableSet());
+    }
+
+    private List<String> anonymizeNames(List<String> names)
+    {
+        return names.stream()
+                .map(this::anonymize)
+                .collect(toImmutableList());
+    }
+
+    private Symbol anonymize(Symbol symbol)
+    {
+        return new Symbol(anonymize(symbol.getName()));
+    }
+
+    private String anonymize(String name)
+    {
+        return anonymizedSymbols.computeIfAbsent(name, key -> String.valueOf(name.hashCode()));
+    }
+
+    private List<Expression> anonymizeExpressions(List<Expression> expressions)
+    {
+        return expressions.stream()
+                .map(this::anonymize)
+                .collect(toImmutableList());
+    }
+
+    private Expression anonymize(Expression expression)
+    {
+        return ExpressionTreeRewriter.rewriteWith(
+                new ExpressionRewriter<>()
+                {
+                    @Override
+                    public Expression rewriteSymbolReference(SymbolReference node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+                    {
+                        return anonymize(Symbol.from(node)).toSymbolReference();
+                    }
+
+                    @Override
+                    public Expression rewriteLiteral(Literal node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+                    {
+                        if (node instanceof StringLiteral) {
+                            return new StringLiteral(anonymize(((StringLiteral) node).getValue()));
+                        }
+
+                        return node;
+                    }
+                },
+                expression);
+    }
+
+    public static class AnonymizedTableHandle
+            implements ConnectorTableHandle
+    {
+        private final QualifiedObjectName tableName;
+        private final String className;
```

Comment:
Should be connectorName after https://github.com/trinodb/trino/pull/12345/files#r871282206